src/dialer/index.js

```diff
@@ -70,7 +70,7 @@ class Dialer {
   async connectToPeer (peer, options = {}) {
     const dialTarget = this._createDialTarget(peer)
 
-    if (!dialTarget.addrs.length) {
+    if (!dialTarget.addrs || !dialTarget.addrs.length) {
```

Comment:
What do you think on doing the following in the `this._createDialTarget`?

```js
let addrs = this.peerStore.addressBook.getMultiaddrsForPeer(id) || []
```

I think it would be cleaner.